cmd/imapmemserver/main.go

```diff
@@ -68,8 +68,9 @@ func main() {
 			return memServer.NewSession(), nil, nil
 		},
 		Caps: imap.CapSet{
-			imap.CapIMAP4rev1: {},
-			imap.CapIMAP4rev2: {},
+			imap.CapIMAP4rev1:                 {},
+			imap.CapIMAP4rev2:                 {},
+			imap.Cap("APPENDLIMIT=104857600"): {},
```

Comment:
I'd prefer to add a new `Session` interface instead of making the servers responsible for crafting this string. This would avoid hardcoding strings and would allow the server to customize the limit depending on the connection.

```go
// SessionAppendLimit is an IMAP session which has the same APPEND limit for
// all mailboxes.
type SessionAppendLimit interface {
	Session
	AppendLimit() uint32
}
```